### PR TITLE
Fix debug symbols 

### LINKED ISSUE #31639: addressToLine misbehaves in master
Note filenames in the backtrace

```
tail -10  clickhouse-server.err.log
5. DB::TCPHandler::receiveHello() @ 0x131f2503 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
6. DB::TCPHandler::runImpl() @ 0x131ebd82 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
7. DB::TCPHandler::run() @ 0x13200499 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
8. Poco::Net::TCPServerConnection::start() @ 0x15e8358f in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
9. Poco::Net::TCPServerDispatcher::run() @ 0x15e85981 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
10. Poco::PooledThread::run() @ 0x15f9c1c9 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
11. Poco::ThreadImpl::runnableEntry(void*) @ 0x15f99900 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
12. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
13. __clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
 (version 21.12.1.8761 (official build))

```
Playing with 00974_query_profile.sql I see the same

```
select addressToLine(toUInt64(293109982));

SELECT addressToLine(toUInt64(293109982))

Query id: 7d7353c9-f4f7-4f81-8d96-3e9e2b6defc6

┌─addressToLine(toUInt64(293109982))───────────────────────────────────────┐
│ /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug │
└──────────────────────────────────────────────────────────────────────────┘


```

Unfortunately, the test does not treat this as failure because condition 'symbol LIKE '%Source%';' is true for some records

It might be that the issue caused by something strange at my box. 
Please, let me know if the issue is reproducible (or not).

### LINKED ISSUE #36797: introspection functions don't load debug symbols from clickhouse-common-static-dbg anymore

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix debug symbols

### Patchset

Debug symbols had been broken in many places and should not work for a long time, the fix includes the following patches:
- Fix debug symbols (Add a quirk to force clang emit .debug_aranges with ThinLTO)
  _Also note, that to support it without a quirk, patch for llvm is required, the discussion can be found [here](https://discourse.llvm.org/t/clang-does-not-produce-full-debug-aranges-section-with-thinlto/64898) and the patch [here](https://reviews.llvm.org/D133092)_
- Fix loading external symbols (previously it does not look at `/usr/lib/debug/usr/bin/clickhouse.debug`)
- Do not compress debug symbols since dwarf parser cannot handle it, this increases the **debug file from 510MB to 1.8GB**, but deb packages compressed anyway, but the size of the **deb** also slightly increased from **834M** to **962M**
- Do not try to load empty debug files (this is required due to build creates empty files for sanitizers and debug builds)
- And also it disables the `02161_addressToLineWithInlines` test, since they do not work always for now (will take a look at this)

Fixes: #36797
Fixes: #31639

### CODE REVIEW

**Comment by azat** [comment]
Everything works as expected, and just for the info:

```
# clickhouse-binary is the binary w/o -flto=thin
$ objdump -j .debug_aranges -h clickhouse-binary 

clickhouse-binary:     file format elf64-x86-64

Sections:
Idx Name          Size      VMA               LMA               File off  Algn
 35 .debug_aranges 005e92e0  0000000000000000  0000000000000000  622f814e  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
```

```
azat@s4:~/ch/tmp/40873$ objdump -j .debug_aranges -h clickhouse

clickhouse:     file format elf64-x86-64

Sections:
Idx Name          Size      VMA               LMA               File off  Algn
 34 .debug_aranges 0055eb30  0000000000000000  0000000000000000  4b52c259  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
```

So size more or less the same.

**Comment by azat** [comment]
Now it works, but inlines does not, I did not managed to fix it quickly so for now `02161_addressToLineWithInlines` had been disabled and in the meantime I will take a look.

Test failures does not looks related and I will take a look in details a little bit later.

File: tests/queries/0_stateless/02161_addressToLineWithInlines.sql [hunk 1]
```diff
@@ -1,4 +1,5 @@
--- Tags: no-tsan, no-asan, no-ubsan, no-msan, no-debug, no-cpu-aarch64
+-- Tags: no-tsan, no-asan, no-ubsan, no-msan, no-debug, no-cpu-aarch64, disabled
+-- Tag disabled: Parsing inlines may lead to "could not find abbreviation code" (FIXME)
```
Comment (novikd) [inline tests/queries/0_stateless/02161_addressToLineWithInlines.sql:2]: Do you plan to continue working on it?

**Review by novikd — APPROVED** [review]
In general, LGTM, but I think it might be useful to create issues for inlines and `--compress-debug-sections` because it's easy to forget that we are missing it.

File: tests/queries/0_stateless/02161_addressToLineWithInlines.sql (hunk 1 above)
Comment (azat) [reply tests/queries/0_stateless/02161_addressToLineWithInlines.sql:2]: Yes, I have this in my todo list (and not the long term one)

**Comment by azat** [comment]
> I think it might be useful to create issues for inlines and --compress-debug-sections because it's easy to forget that we are missing it.

Actually I was going to take a look at the llvm parser (folly does not support it still), and I guess it will support them automatically.

However I'm not sure that this worth it, because symbolizing stacktraces takes lots of time here (and hence have a cache), and increasing it even more does not looks good to me, anyway deb packages are compressed (so debug file will be compressed anyway), and regular binaries compressed with zstd now.

But robust DWARF parser is still required, and maybe decompression will not take that much...

**Comment by azat** [comment]
Ok, #41000, #35405 and maybe #40772 should be enough to improve the situation

**Comment by azat** [comment]
I would prefer to merge this before #41046 (it may help with the `00974_query_profiler` failures there and also maybe clang-15 will make something different and this will stop working again).

**Comment by novikd** [comment]
> Stress test (msan) — Killed by signal (in clickhouse-server.log)

Must be fixed in #41050

> Stateless tests (release, s3 storage) — fail: 1, passed: 4364, skipped: 61
> Stateless tests (ubsan) — Tests are not finished, fail: 1, passed: 90, skipped: 3

Seems unrelated

**Pull request merged**